front/src/modules/simulationResult/components/SpaceTimeChartWrapper/useTrackOccupancy.ts

```
@@ -238,40 +238,56 @@ const useTrackOccupancy = ({
-            if (!train || isTrainScheduleProjection(train)) {
+
+            if (!train) throw new Error(`No train found for id ${pacedId}`);
+            if (isTrainScheduleProjection(train))
+              throw new Error(`TrainSchedule projection found for id ${pacedId}`);
+
+            if (occupation.type === 'base') {
+              zones.push(
+                getMovableOccupancyZone(
+                  trackId,
+                  pacedId,
+                  occupation,
+                  train.spaceTimeCurves,
+                  train.name,
+                  train.departureTime
+                )
+              );
               continue;
             }
 
-            const exception =
-              occupation.type === 'BaseOccurrence'
-                ? undefined
-                : train.paced.exceptions.find((e) => e.key === occupation.exception_key);
+            if (!train.paced) throw new Error(`Train with id ${train.id} is not a paced train`);
 
-            const exceptionProjection =
-              occupation.type !== 'BaseOccurrence'
-                ? train.paced.exceptionProjections.get(occupation.exception_key)
-                : undefined;
+            const exception = train.paced.exceptions.find(
+              (e) => e.key === occupation.exception_key
+            );
+            const exceptionProjection = train.paced.exceptionProjections.get(
+              occupation.exception_key
+            );
 
             const { spaceTimeCurves } = exceptionProjection ?? train;
 
             let trainId: TrainId;
             let trainName: string;
             let startTime: Date;
 
-            if (occupation.type === 'CreatedException') {
-              if (!exception?.start_time?.value) continue;
+            if (!exception) throw new Error(`Exception not found for train ${train.id}`);
+
+            if (occupation.type === 'created') {
+              if (!exception.start_time?.value) continue;
```

A created exception necessarily has a start time. Since a bunch of invalid states were turned into errors above, should we also throw an exception here?

---

Done

---

Done